plugins/auth-backend/src/providers/cloudflare-access/provider.ts

```diff
@@ -0,0 +1,345 @@
+    res.json(response);
+  }
+
+  start(): Promise<void> {
+    return Promise.resolve();
+  }
+
+  private async getIdentityProfile(
+    jwt: string,
+  ): Promise<CloudflareAccessIdentityProfile> {
+    const headers = new Headers();
+    // set both headers just the way inbound responses are set
+    headers.set(CF_JWT_HEADER, jwt);
+    headers.set('cookie', `${COOKIE_AUTH_NAME}=${jwt}`);
+    try {
+      const res: FetchResponse = await fetch(
+        `https://${this.teamName}.cloudflareaccess.com/cdn-cgi/access/get-identity`,
+        { headers },
+      );
+      if (!res.ok) {
+        // Cast to work around https://github.com/backstage/backstage/issues/12166
+        throw await ResponseError.fromResponse(res as unknown as Response);
+      }
+      const cfIdentity = await res.json();
+      return cfIdentity as unknown as CloudflareAccessIdentityProfile;
+    } catch (err) {
+      this.logger.error(`getIdentityProfile failed: ${err}`);
+      return Promise.reject(err);
+    }
+  }
+
+  private async getResult(
+    req: express.Request,
+  ): Promise<CloudflareAccessResult> {
+    // JWTs generated by Access are available in a request header as
+    // Cf-Access-Jwt-Assertion and as cookies as CF_Authorization.
+    let jwt = req.header(CF_JWT_HEADER);
+    if (!jwt) {
+      jwt = req.cookies.CF_Authorization;
+    }
+    if (!jwt) {
+      // Only throw if both are not provided by Cloudflare Access since either
+      // can be used.
+      throw new AuthenticationError(
+        `Missing ${CF_JWT_HEADER} from Cloudflare Access`,
+      );
+    }
+
+    // Cloudflare signs the JWT using the RSA Signature with SHA-256 (RS256).
+    // RS256 follows an asymmetric algorithm; a private key signs the JWTs and
+    // a separate public key verifies the signature.
+    const verifyResult = await jwtVerify(jwt, this.jwtKeySet, {
+      issuer: `https://${this.teamName}.cloudflareaccess.com`,
+    });
+    const cfAccessResultStr = await this.cache?.get(jwt);
+    if (typeof cfAccessResultStr === 'string') {
+      return JSON.parse(cfAccessResultStr) as CloudflareAccessResult;
+    }
+    const claims = verifyResult.payload as CloudflareAccessClaims;
+    // Builds a passport profile from JWT claims first
+    const cfAccessResult: CloudflareAccessResult = {
+      claims,
+      expiresInSeconds: claims.exp - claims.iat,
+    };
+    try {
+      // If we successfully fetch the get-identity endpoint,
+      // We supplement the passport profile with richer user identity
+      // information here.
+      const cfIdentity = await this.getIdentityProfile(jwt);
+      cfAccessResult.cfIdentity = cfIdentity;
+      // Stores a stringified JSON object in cfaccess provider cache only when
+      // we complete all steps
+      this.cache?.set(jwt, JSON.stringify(cfAccessResult));
+    } catch (err) {
+      this.logger.error(
+        `failed to populate access identity information: ${err}`,
+      );
```

Comment:
It feels a bit odd to consider this an optional fetch tbh. Either the identity resolution logic is gonna need the profile information, or it won't? Perhaps we could make the entire profile fetch configurable instead, so that it only happens if you configure the auth provider to execute the fetch? Thinking something like this:

```ts
providers.cfAccess.create({
  shouldFetchIdentityProfile: true,
  signIn: {
    async resolver({ profile, result }, ctx) {
      ...
    },
  },
})
```

Reply:
Fair enough, I'll update it so when get-identity fail, it just throws